li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/largemessage/ConsumerRecordsProcessor.java

```diff
@@ -457,14 +471,20 @@ public long getConsumerOffsetWatermarkSpan() {
     // Create a new copy of the headers
     Headers headers = new RecordHeaders(consumerRecord.headers());
     Header largeMessageHeader = headers.lastHeader(Constants.LARGE_MESSAGE_HEADER);
+    Header encryptionHeader = headers.lastHeader(Constants.ENCRYPTION_HEADER);
```

Comment:
This header can be read once the entire assembled record is available. So, this can move below inside the `if (valueBytes != INCOMPLETE_RESULT)` clause

Reply:
Yes. Updated.